### PR TITLE
Endpoint: keys/check

### LINKED ISSUE #31: Adding 'keys/check' api endpoint

### PR DESCRIPTION
**Pull request by Mikkel-Boje**
Closes #31
Point of this endpoint is to check if the client and server has the same keys for the client.

### CODE REVIEW

File: common/src/web_api.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ pub struct CreateAccountOptions {
 #[derive(Debug, Serialize, Deserialize, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct UploadSignedPreKey {
-    pub key_id: u32,
+    pub key_id: u64,
```
Comment (AOsnes) [inline common/src/web_api.rs:27]: ensure the client can handle a u64 as the id

File: server/src/server.rs
Comment (AOsnes) [inline server/src/server.rs:1]: I would like some testing of this file 😄

**Review by AOsnes — CHANGES_REQUESTED** [review]
Please implement feedback

File: common/src/web_api.rs (hunk 1 above)
Comment (Mikkel-Boje) [reply common/src/web_api.rs:27]: This is defined by by the specification therefore it needs to be 8 bytes long.

File: server/src/server.rs
Comment (Mikkel-Boje) [reply server/src/server.rs:1]: Testing is done :)

File: common/src/web_api.rs (hunk 1 above)
Comment (Mikkel-Boje) [reply common/src/web_api.rs]: Found out that libsignal doesnt even support using an 8 byte variant. Therefore i just added a comment such that we are aware of this difference that our design could have compared to theirs.

File: server/src/server.rs [hunk 2]
```diff
@@ -18,10 +21,33 @@ type UserID = u32;
...
+    Kem(kem::PublicKey),
+    Ec(PublicKey),
+}
+
+impl PublicKeyType {
+    fn expect_kem(self) -> kem::PublicKey {
+        if let PublicKeyType::Kem(key) = self {
+            key
+        } else {
+            panic!()
```
Comment (MJHC) [inline server/src/server.rs:34]: write some panic message

File: server/src/server.rs [hunk 3]
```diff
@@ -18,10 +21,33 @@ type UserID = u32;
...
+        }
+    }
+    fn expect_ec(self) -> PublicKey {
+        if let PublicKeyType::Ec(key) = self {
+            key
+        } else {
+            panic!()
```
Comment (MJHC) [inline server/src/server.rs:41]: also here

**Review by MJHC — CHANGES_REQUESTED** [review]
make panic msgs

**Review by AOsnes — DISMISSED** [review]
👍🏻

File: server/src/server.rs [hunk 4]
```diff
@@ -18,10 +21,33 @@ type UserID = u32;
...
+            key
+        } else {
+            panic!("dev_err: expected a kem key, got an ec")
+        }
+    }
+    fn expect_ec(self) -> PublicKey {
+        if let PublicKeyType::Ec(key) = self {
+            key
+        } else {
+            panic!("dev_err: expected an ec key, go kem")
```
Comment (MJHC) [inline server/src/server.rs:41]:
```suggestion
            panic!("dev_err: expected an ec key, got kem")
```

**Review by MJHC — CHANGES_REQUESTED** [review]
go kem?

**Review by MJHC — APPROVED** [review]
ok

**Review by AOsnes — APPROVED** [review]
♿

**Pull request merged**